spring-web/src/main/java/org/springframework/web/multipart/support/DefaultMultipartHttpServletRequest.java

```diff
@@ -87,14 +89,28 @@ public String getParameter(String name) {
 		}
 		return super.getParameter(name);
 	}
+	
+	public String[] mergeTwoParameterValues(String[] val1, String[] val2) {
+		if (val1 == null) {
+			return val2;
+		}
+		if (val2 == null) {
+			return val1;
+		}
+		//merge them together before return
+		String[] merged = Arrays.copyOf(val2, val1.length + val2.length);
```

Comment:
It's counter-intuitive for a method that merges to arrays, to put the second one first.

Reply:
So i fix them and add getParameterValues test in CommonsMultipartResolverTests.

But i used forced pushing to avoid unnecessary commits and found i cannot reopen this.

Should i open another PR?

Reply:
Force push is fine. Just keep the branch updated, no need to re-create the PR.